src/warc2zim/converter.py

```diff
@@ -282,17 +283,24 @@ def iter_all_warc_records(self):
 
         yield from iter_warc_records(self.inputs)
 
-    def find_main_page_metadata(self):
-        for record in self.iter_all_warc_records():
+    def gather_information_from_warc(self):
```

Comment:
I think the name is misleading. It's all about the main page (and its metadata) and given how long this is already, probably good not to add additional stuff to it

Reply:
The problem is that doing the main page searching and the filling of known_url in two different steps means that we loop twice (and one more time for the real processing) the warc entries.

Reply:
I am not suggesting we change the behavior ; just the name. Maybe `find_homepage_and_urls` or something